wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java

```diff
@@ -571,6 +787,31 @@ public void removeSnapshots(String workspaceId) throws NotFoundException, Server
     }
 
     /**
+     * Removes all snapshots of all machines of specified environment of workspace.
+     *
+     * @param workspaceId
+     *         workspace id to remove environment snapshots
+     * @param envName
+     *         environment name to remove snapshots from
+     * @throws NotFoundException
+     *           when workspace with given id doesn't exists
+     * @throws ServerException
+     *          when any other error occurs
+     */
+    public void removeEnvironmentSnapshots(String workspaceId, String envName) throws NotFoundException, ServerException {
+        getSnapshot(workspaceId).stream()
+                                .filter(snapshot -> snapshot.getEnvName().equals(envName))
```

Comment:
Is it probable to find more than 1 entry? If not then filter doesn't suit this situation well